uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/components/leave-type-wizard.component.js

```diff
@@ -64,6 +64,9 @@ define([
         .then(function () {
           tabsIndexed = indexPropertyByName(vm.sections, 'tabs');
           vm.fieldsIndexed = indexPropertyByName(tabsIndexed, 'fields');
+          promises.titlesInUse = fetchAbsenceTypesTitlesInUse();
+
+          promises.titlesInUse.then(validateTitleField);
```

Comment:
this logic doesn't seem to flow naturally. Why would you validate the field after the titles have been fetched? One necessarily doesn't follow the other, as the field should be validated only when the its value changed, not when the titles have loaded

For example: if i go to the wizard page and then do nothing for the next 30 seconds, why should the wizard try to validate the title at all? All you are supposed to be doing here is to fetch the data that will be used for the validation _at a later time_

Reply:
Yes, I am not sure why I added this, good catch. Thanks.